docs: introduction.md changes (#4393)
botpress/botpress: docs/guide/docs/introduction.md

title: Introduction
---

Building chatbots is a new, exciting challenge for most developers. Creating software that holds a conversation and responds naturally to humans is a thrilling possibility, and Botpress is turning this into reality. 

> ![Simple Pizza Bot](assets/pizzabot.gif)

Reviewer: I don't think we need a screenshot at this point

Reviewer 2: Agree, remove the screenshot

Author: Alright

> The catch is that you (or anyone else) wouldn't want to spend hundreds of hours learning and putting together all the infrastructure and boilerplate code required to get a conversational backend up and running.

Reviewer 2: ```suggestion
The problem is that you wouldn't want to spend hundreds of hours learning and putting together all the infrastructure and boilerplate code required to get a conversational backend up and running. Luckily, you don't have to.
```

## What is Botpress

> > A **lightweight**, **fast**, and **flexible** chatbot building framework that you can host anywhere, whether it be on the cloud or your servers.

Reviewer: ```suggestion
> A **lightweight**, **fast**, and **flexible** chatbot building framework that you can host anywhere, whether it be on the cloud or on-premise.
```

> > A **lightweight**, **fast**, and **flexible** chatbot building framework that you can host anywhere, whether it be on the cloud or your servers.

Reviewer 2: ```suggestion
> A **lightweight**, **fast**, and **flexible** chatbot building platform that you can host on the cloud or on-premise.
```

> It is a complete platform that ships with all the tools you need to build, deploy, and manage production-grade bots in record time. Here are some of the components that are installed by default:

Reviewer 2: ```suggestion
It is a complete platform that ships with all the tools you need to build, deploy and manage production-grade bots in record time. Here are some of the components that are installed by default:
```

> - An NLU Engine which understands the meaning and intentions behind user statements even where grammatical mistakes exist.

Reviewer 2: ```suggestion
- An **NLU Engine** for intent detection and variable extraction.
```

> - An administration interface that can be viewed from a browser, exposing modules to manage content and show chatbot  statistics, among other things. 

Reviewer 2: ```suggestion
- An **administration interface** that can be viewed from a browser, to manage and monitor your instances, bots and users.
```

> - A visual workflow builder used for creating conversations and other interactions.

Reviewer 2: ```suggestion
- A visual **workflow builder** to create conversation flow and workflow automations.
```

> - A chat emulator/debugger to simulate conversations and inspect the chatbot engine's logic.

Reviewer 2: ```suggestion
- A **chat emulator/debugger** to simulate conversations and debug your chatbot.
```

> - Support for popular messaging channels like Facebook Messenger, Slack, Telegram, Microsoft Teams, and others.

Reviewer 2: ```suggestion
- Support for **popular messaging channels** like Facebook Messenger, Slack, Telegram, Microsoft Teams, and others.
```

## Why Botpress

Botpress has many advantages over other platforms and frameworks:

> - It runs **fully on-premise**, so you can host it on a server of your choice. This gives you full control over the data that comes in and out and helps you secure your data. 

Reviewer 2: ```suggestion
- It runs **fully on-premise**, so you can host it on a server of your choice, giving you full control over the data that comes in and out.
```

> The Botpress platform's core is **open-source** with thousands of Github contributors and will be free forever.

Reviewer: ```suggestion
The Botpress platform's core is **open-source** with thousands of GitHub contributors and will be free forever.
```

> The Botpress platform's core is **open-source** with thousands of Github contributors and will be free forever.

Reviewer 2: ```suggestion
The Botpress platform's core is **open-source** with thousands of GitHub contributors.
```

> - The framework is very **flexible** and developer-friendly because Botpress can be customized primarily using Javascript with capabilities to include code in other languages.

Reviewer 2: ```suggestion
- The platform is extremely **flexible** and developer-friendly.
```

> - Botpress relies on **zero** third-party services, making it the only fully featured Chatbot-Building platform

Reviewer 2: ```suggestion
- **Zero** external dependencies, making it the only fully featured chatbot building platform
```

> ## How to build a Chatbot

Reviewer: ```suggestion
## How to build a chatbot
```

There are three main steps involved in creating chatbots.

Developers start by **Building** a chatbot, which consists of adding the required messaging channels, integrating the necessary backend components and applications your chatbot will need to interact with, and creating conversation workflows.

> Once done, you'll need to **Deploy** your chatbot somewhere, which could be on a cloud server or your internal hosting infrastructure.

Reviewer: ```suggestion
Once done, you'll need to **Deploy** your chatbot somewhere, which could be on a cloud server or on-premise.
```

## Why Botpress

Botpress has many advantages over other platforms and frameworks:

- It runs **fully on-premise**, so you can host it on a server of your choice. This gives you full control over the data that comes in and out and helps you secure your data. 
The Botpress platform's core is **open-source** with thousands of Github contributors and will be free forever.
- The framework is very **flexible** and developer-friendly because Botpress can be customized primarily using Javascript with capabilities to include code in other languages.
- Botpress relies on **zero** third-party services, making it the only fully featured Chatbot-Building platform

## How to build a Chatbot

There are three main steps involved in creating chatbots.

Developers start by **Building** a chatbot, which consists of adding the required messaging channels, integrating the necessary backend components and applications your chatbot will need to interact with, and creating conversation workflows.

Once done, you'll need to **Deploy** your chatbot somewhere, which could be on a cloud server or your internal hosting infrastructure.

> Finally, you'll give other people access to your chatbot so they can **Manage** it once it is in use. Managing a chatbot entails analyzing the conversations, refining the conversational experience by building new flows, and making it smarter by continuously improving the NLU.

Reviewer 3: you use workflows instead of flows pretty much everywhere else. for a matter of consistency let's keep it as workflow
